### ISSUE TITLE
Solution view shows duplicated items

### ISSUE THREAD

**Issue by drasticactions** [mono/monodevelop#5964]
VSMac Version: 7.6.1.10 (Also tested with 7.8.0.408)

<img width="193" alt="screen shot 2018-09-13 at 9 26 08 pm" src="https://user-images.githubusercontent.com/898335/45524283-a6bcb900-b79b-11e8-9e8a-30c6f0514f4f.png">

For some workloads, items are appearing twice inside of the solution view. Clicking on either item in the list will show the editor for that item. It does not affect the csproj itself; it seems to be cosmetic only. I've only been able to replicate it so far with ASP.NET Core workloads, specifically when using the SPA templates. It may affect other items, but I'm not sure.

Steps to Replicate:

```
mkdir Test
cd Test
dotnet new react
```

This should create an ASP.NET workload with a `create-react-app` SPA.

Launch the `Test.csproj` in VS For Mac, and view the dropdown, you'll see the multiple items listed in `ClientApp`. The other folders and root are fine. Moreover, if you copy one of the duplicated items out of `ClientApp`, it will only appear only once.

**Comment by mrward** [comment]
There is a bug in the ASP.NET Core project templates where they add some files in the ClientApp as None items but do not remove the None items defined by default by the .NET Core SDK. VS Mac does not hide duplicate MSBuild items that are defined by the project.

If you edit the .csproj and change the ItemGroup to include `<None Remove="$(SpaRoot)**" />` then the duplicate files should be fixed:

```
<ItemGroup>
  <!-- Don't publish the SPA source files, but do show them in the project files list -->
  <Content Remove="$(SpaRoot)**" />
  <None Remove="$(SpaRoot)**" />
  <None Include="$(SpaRoot)**" Exclude="$(SpaRoot)node_modules\**" />
</ItemGroup>
```

https://github.com/aspnet/Templating/issues/559

**Comment by thejones** [comment]
Out of curiosity, I started a project today and noticed this. I do not see a .csproj file in the new template. Is there another fix or workaround that I should try? On a Mac and used the CLI to create a new project.

**Comment by mrward** [comment]
@thejones Would need to know what sort of project is being used. I would open another github issue with more information? Thanks.

**Comment by enRose** [comment]
I started off a new react-redux template from vs 2019 for mac showing duplicated files on clientApp.

**Comment by localhost09** [comment]
@enRose  can confirm the same problem

**Comment by Vectrino314** [comment]
Hi guys, if any of you are facing the same problem, your .csproj file is actually the root project so open it in an external code editor and you will be able to copy the code <None Remove="$(SpaRoot)**" /> and it will work. It just fixed it for me.
<img width="599" alt="Root" src="https://user-images.githubusercontent.com/35986802/64047480-831d8a80-cb34-11e9-815e-edd7541ebb8b.png">

**Comment by netpoetica** [comment]
I'm not sure if this is the right place to mention this, but I get the exact same problem using the gRPC templates in Visual Studio for Mac following these instructions: https://docs.microsoft.com/en-us/aspnet/core/tutorials/grpc/grpc-start?view=aspnetcore-3.1&tabs=visual-studio-mac

<img width="331" alt="Screen Shot 2020-01-09 at 10 10 39 AM" src="https://user-images.githubusercontent.com/1972316/72079757-82a6b480-32c9-11ea-933b-af51944f90e4.png">

Let me know if you prefer that I write a new ticket or post somewhere else. Cheers!

**Comment by mrward** [comment]
@netpoetica This is a known bug in VS Mac if you have .NET Core 3.1 sdk installed.

This has been filed on the Developer Community site a few times - https://developercommunity.visualstudio.com/content/problem/842829/net-core-web-application-mvc-template-creates-mult.html

**Comment by netpoetica** [comment]
Thanks - in my case, it isn't just the settings files though. You will see there is a duplicate .proto file as well. I imagine it is the same issue, but worth mentioning that it isn't just settings

**Comment by mrward** [comment]
For .proto files it looks like it is a problem with the default None items not being removed. That seems to be due to the imported MSBuild props defined by Grpc not removing them. VS Mac currently does not hide MSBuild items that are included twice by the sdk. This is different to the problem on the developer community but I think the plan will be to fix both these problems so VS Mac is consistent with VS Win.

A workaround for the duplicate .proto file is to add a None remove inside an ItemGroup in the csproj file:

    <ItemGroup>
      <None Remove="**\*.proto" />
      <Protobuf Include="Protos\greet.proto" GrpcServices="Server" />
    </ItemGroup>
